### PR TITLE
save 301 responses from webcal subscriptions

### LINKED ISSUE #42: Update webcal url stored in caldav when it is a 301

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #42

### CODE REVIEW

File: controller/proxycontroller.php [hunk 1]
```diff
@@ -76,14 +76,60 @@ public function __construct($appName, IRequest $request,
 	 */
 	public function proxy($url) {
 		$client = $this->client->newClient();
+		//$history = new GuzzleHttp\
```
Comment (georgehrke) [inline controller/proxycontroller.php]: - [ ] remove

File: controller/proxycontroller.php [hunk 2]
```diff
@@ -76,14 +76,60 @@ public function __construct($appName, IRequest $request,
...
+				} else {
+					$done = true;
+				}
+			} while(!$done && $redirect_count <= $max_redirects);
+
+			if ($redirect_count > 0 && $redirect_count <= $max_redirects) {
+				$response = new JSONResponse([
+					'proxy_code' => -4,
+					'new_url' => $queryUrl,
+				], Http::STATUS_IM_A_TEAPOT);
```
Comment (tcitworld) [inline controller/proxycontroller.php]: 303 should do it

Comment (georgehrke) [inline controller/proxycontroller.php]: Nope, I thought about using a 3xx code. But the browser will expect a Location header and just forward there.

Comment (tcitworld) [inline controller/proxycontroller.php]: 400 can handle this situation.

**Comment by georgehrke** [comment]
please review @tcitworld

**Pull request merged**